src/sentry/api/endpoints/project_create_sample_transaction.py

```diff
@@ -0,0 +1,44 @@
+import os
+from datetime import datetime, timedelta
+
+import pytz
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint, ProjectEventPermission
+from sentry.api.serializers import serialize
+from sentry.constants import DATA_ROOT
+from sentry.utils import json
+from sentry.utils.dates import to_timestamp
+from sentry.utils.samples import create_sample_event_basic
+
+
+class ProjectCreateSampleTransactionEndpoint(ProjectEndpoint):
+    # Members should be able to create sample events.
+    # This is the same scope that allows members to view all issues for a project.
+    permission_classes = (ProjectEventPermission,)
+
+    def post(self, request, project):
+        samples_root = os.path.join(DATA_ROOT, "samples")
+        with open(os.path.join(samples_root, "react-transaction.json")) as fp:
+            data = json.load(fp)
+
+        timestamp = datetime.utcnow() - timedelta(minutes=1)
```

Comment:
@Neo-Zhixing I see a lot this code is copied from: https://github.com/getsentry/sentry/blob/master/src/sentry/utils/samples.py#L170-L212

Anyway we could make this into a function then call it in this endpoint?

Reply:
This is more like a special case compared to https://github.com/getsentry/sentry/blob/master/src/sentry/utils/samples.py#L170-L212 as it only handles time-related stuff. I would be inclined to make this thing self-contained, because 1. it's not much code and 2. we can iterate in the future if we see the needs.

Reply:
@Neo-Zhixing yep, thought as much. What you have is good then.